tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -41,12 +42,25 @@
 
 
 class TemporalMemoryTest(unittest.TestCase):
+  @staticmethod
+  def isSegmentDestroyed(connections, segment):
+    try:
+      connections.synapsesForSegment(segment)
+      return False
+    except KeyError:
+      return True
 
+  @staticmethod
+  def isSynapseDestroyed(connections, synapse):
+    try:
+      connections.dataForSynapse(synapse)
+      return False
+    except KeyError:
+      return True
 
   def setUp(self):
     self.tm = TemporalMemory()
```

Comment:
This is now mostly unused. Only a couple tests use it, but every test pays the cost of it. It's probably worth moving this line into the three tests that use `self.tm` and removing this `setUp` method.